app/src/main/java/com/orgzly/android/data/DataRepository.kt

```diff
@@ -1597,13 +1607,30 @@ class DataRepository @Inject constructor(
     }
 
     private fun tryUpdateTitleCookies(note: Note) {
+        // as of my testing in org 9.6.15, if a heading has child headings which are TODO items
+        // as well as content which are checkboxes, it's unpredictable how the title changes, but:
+        //
+        // 1. it does NOT count both the checkboxes and child headings together, but rather favors
+        //    one or the other
+        // 2. it seems to favor them in the order they appear (top-down), but also somehow seems to
+        //    favor child headings more
+        //
+        // for sanity's sake, we're just going to process for any checkboxes and then process for
+        // any children, in that order
```

Comment:
Good thinking, IMO. We should always try to stick as close to org-mode's default behavior as possible. And we don't parse the COOKIE_DATA property. (We could add support for that, but it gets us into bigger questions around parsing settings/properties in headings and files in general.)